wizard/views/record_sets.py

```diff
@@ -11,31 +12,29 @@
 
 
 def render_record_sets():
-    if not st.session_state[RecordSets]:
-        st.markdown("Provide files before.")
+    if len(st.session_state[Metadata].record_sets) == 0:
```

Comment:
Prefer `if not list` (https://google.github.io/styleguide/pyguide.html#2144-decision).

Reply:
I understand that it's in the (google) style guide, but that is a terrible style/readability decision, it forces readers to figure out why it works (if they don't know), rather than relying on the plainly readable above.  Python styling/readability decisions (regardless of the source) seem to want to eschew readability to add "pythonicity" which functionally just means arcane decisions of a specific type (almost always arbitrary or involved in using language specific features, seemingly only to justify their continued existence).  Even the style guide you just linked has recommendations but no justifications for *why* this should be done, especially when it makes this flatly less readable. I generally have no problem running automated linters over code (as developer time spent is tiny). However, I would prefer not getting into these conversations as we're under time limitations that make such arbitrary style nits a tertiary concern at best, in my opinion.

Reply:
apologies if this was a bit of a rant, I've had python developers give "pythonic code" the same level of adherence as code structure or nominal functional code boundaries (things which do help readability, long term code health, and readability) and i'm always puzzled as to the value of writing code this way.  To date, none of them have been able to explain *why* it has value, in much the same way that golang developers can't explain why golang's "simplicity" is a positive trait.

Reply:
Thanks for expressing your opinion and the detailed comments :)

The *why* justification of the rule is in the style guide I linked ([link](https://google.github.io/styleguide/pyguide.html#2142-pros)):

```
Conditions using Python booleans are easier to read and less error-prone. In most cases, they’re also faster.
```

This style guide is the one we've used so far, which is why I linked it. For example, reading through it can help you understand why the line used to be `if not st.session_state[RecordSets]:`.

I'd like us to stick to homogeneous rules within the code base. The Google style guide happens to be the style we picked since the beginning. As we start the editor from the existing codebase of mlcroissant, I guess it's just easier to stick with the existing style. :) WDYT?

Reply:
I understand the desire to follow the style guide, but I would be remiss if I didn't point out the folly of adhering to a style guide while a project deadline looms this close. We should talk about this on Thursday, but I'm a little concerned with the desire for quality over the desire for execution, and what that means for what we can accomplish before the end of November.  I'm not sure it's a great use of my time to devote this much effort into making each PR as high of a quality as you are clearly used to.   I would much prefer to generate a bit of tech debt/messy code now and get much more done feature wise, I certainly have no problem addressing this after the end of November (or even before, if we get most of our desired features done), even if I am no longer under contract, but don't feel like this is a great use of the time we have now.  In addition, it seems like some of these nits are getting in the way of discovering actual bugs, as I've noticed and caught a couple that have been completely missed.

Reply:
I hope to achieve both quality and execution! Changing a line of code shouldn't take too much time. We can discuss on Thursday to see where we put the bar.

The point on catching bugs is a good point... From now on, I'll checkout the branch locally to have a closer look and hopefully catch more bugs. Thanks!